propka/atom.py

```diff
@@ -290,8 +290,9 @@ def get_input_parameters(self):
             self.occ = self.occ.replace('LG', 'non_titratable_ligand')
             # try to initialise the group
             try:
-                # TODO - get rid of this exec() statement for security reasons
-                exec('self.group = propka.group.%s_group(self)' % self.occ)
+                group_attr = "%s_group" % self.occ
+                group_attr = getattr(propka.group, group_attr)
+                self.group = group_attr(self)
```

Comment:
If you bump the required Python version to 3.6, then you can use f-strings:
```python
group_attr = getattr(propka.group, f"{self.occ}_group")
```

Reply:
Oooh... fancy. 😄 

Unfortunately, I still regularly encounter Python 3.5 in places so we might want to hold off for a while?

Reply:
Sure that's fine.